crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -368,6 +368,16 @@ where
                     .add_or_update_symbol(function_def.name.id.clone(), SymbolFlags::IS_DEFINED);
                 self.add_definition(symbol, function_def);
 
+                // The default value of the parameters needs to be evaluated in the
+                // enclosing scope.
+                for default in function_def
+                    .parameters
+                    .iter_non_variadic_params()
+                    .filter_map(|param| param.default.as_deref())
+                {
+                    self.visit_expr(default);
+                }
```

Comment:
nit: you could also do this like this, which feels slightly nicer to me (but no strong opinion):

```suggestion
                for default in function_def
                    .parameters
                    .iter()
                    .filter_map(ast::AnyParameterRef::default)
                {
                    self.visit_expr(default);
                }
```

The same applies to the `Expr::Lambda` branch below as well

Reply:
Thanks for the suggestion. I choose `iter_non_variadic_params` because it makes it explicitly clear that we will never iterate over parameters that can't have default values (also mentioned in the function docs).